### ISSUE TITLE
`Invoke-PSModulePublisher.ps1` does not take params

### ISSUE THREAD

**Issue by leojonathanoh** [theohbrothers/PSModulePublisher#24]
## Current

`Invoke-PSModulePublisher.ps1` does not take params

## Discussion

If `Invoke-PSModulePublisher.ps1` is to be used in the CI (as stated in the readme), it must be able to take params `-Repository` and `DryRun`.

Right now, the script is a standalone script that is to be used interactively.

**Comment by joeltimothyoh** [comment]
`Invoke-PSModulePublisher.ps1` as the was designed for use in dev environments: https://github.com/theohbrothers/PSModulePublisher/blob/aa99891eac2de091060b8435fd23762163bb75e3/src/Invoke-PSModulePublisher.ps1#L2

CI environments on the other hand use the public scripts which are more visible with recent changes: https://github.com/theohbrothers/PSModulePublisher/tree/8673a397484d0939cfd51ad4688ab3af6a62e5c4/src

**Comment by leojonathanoh** [comment]
as described in https://github.com/theohbrothers/PSModulePublisher/issues/25#issuecomment-1484485869 that there is a need for the user to be able to use `Invoke-Test`, `Invoke-Build`, and `Invoke-Publish` independently, instead of `Invoke-PSModulePublisher` which performs all three of them at once. Also discussed there is that it seems like the module is making correct progress with #26, #27, #28 which are making all three public functions.

**Comment by joeltimothyoh** [comment]
The project actually isn't a PS module because it contains scripts and CI templates for use with it as well. It's possible to convert the scripts to functions. Will have to see if it's possible to do so without introducing too many breaking changes.

**Comment by leojonathanoh** [comment]
It is possible. I've checked that all the `Invoke-Build`, `Invoke-Test`, `Invoke-Publish` cmdlets are not used by the default PS cmdlets.

**Comment by leojonathanoh** [comment]
the only difficulty i foresee are that the dot-sourcing in `Invoke-Build.ps1` to `Get-ProjectVariables.ps1` uses relative paths.  If this becomes a PS module `Import-Module; Invoke-Build` for instance, the `Invoke-*` cmdlets must now be executed in the PS module's project dir to be able too resolve variables correctly (via private function `Get-ProjectVariables`), in contrast to now where the `Invoke-*` script may be called from anywhere.  Wherwa

**Comment by leojonathanoh** [comment]
true. basically a tools git submodule with a bunch of helpers. shouldn't need a PSGallery to `Install Module; Import Module` in CI environments.

**Comment by leojonathanoh** [comment]
btw, `pwsh` has become mainstream in at least Github CI, with first class citizen availability: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-powershell

so it's pretty easy to any of these in CI

## PSGallery PS module

```sh
Install-Module myModule  ...
Import-Module myModule -Force # May not be necessary, since Cmdlet might be loaded dynamically
My-Cmdlet ...
```

## Git submodule as PS module

```sh
# Must have git clone --recursive to clone submodules
Import-Module ./path/to/src/PSModulePublisher
Invoke-Build ...
```

## GIt submodule as PS script

```sh
# Must git clone --recursive to clone submodules
./path/to/src/Invoke-Build.ps1
```

just some little ideas.

**Comment by joeltimothyoh** [comment]
How about the installation of publish dependencies, Pester, testing, and publishing of PS modules via actions as documented in the link. I'd assume it'd be pretty seamless. Code would be coupled and dependent on GHA though.

**Comment by leojonathanoh** [comment]
Reviewing things again, the approach https://github.com/theohbrothers/Generate-DockerImageVariants/pull/78 is the most elegant imo. The CI jobs do not execute the monolithic `Invoke-PSModulePublisher` but only the scripts needed.

**Comment by joeltimothyoh** [comment]
Resolved in #46.

Closing.
